src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -143,6 +177,19 @@ def register_submission(
         rol_data = apply_data_mapping(
             submission, self.rol_mapping, REGISTRATION_ATTRIBUTE
         )
+
+        betrokkene_identificatie = rol_data.get("betrokkeneIdentificatie", {})
+        kvk = betrokkene_identificatie.pop("_kvk", None)
+
+        if kvk:
+            rol_data["betrokkeneType"] = "vestiging"
+            # Fall back on KvK-number if no vestigingsnumber.
+            betrokkene_identificatie["vestigingsNummer"] = betrokkene_identificatie.get(
+                "vestigingsNummer", kvk
+            )
```

Comment:
I think this doesn't catch the case where the field has an empty string for value - then there will not be a fallback to `kvk`.